aruco_detect/src/aruco_detect.cpp

```diff
@@ -318,7 +318,7 @@ void FiducialsNode::imageCallback(const sensor_msgs::ImageConstPtr & msg) {
         aruco::detectMarkers(cv_ptr->image, dictionary, corners, ids, detectorParams);
         ROS_INFO("Detected %d markers", (int)ids.size());
 
-        for (int i=0; i<ids.size(); i++) {
+        for (unsigned int i=0; i<ids.size(); i++) {
```

Comment:
```suggestion
        for (size_t i=0; i<ids.size(); i++) {
```

size_t is the actual type being used here, and is the correct size for pointers on the system (32 or 64 bit).